edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Core/DeviceScopeIdentitiesCache.cs

```diff
@@ -206,9 +213,11 @@ public void Dispose()
         {
             this.encryptedStore?.Dispose();
             this.refreshCacheTimer?.Dispose();
-            this.refreshCacheTask?.Dispose();
+            this.lifeCycleSignal.Cancel();
```

Comment:
>this.lifeCycleSignal.Cancel(); [](start = 12, length = 30)

hmm...this just signals cancel but doesn't wait.  What happens when background task gets disposed now?

Reply:
The background task will finish by itself. It could wait `refreshCacheTask` finish here if would like to guarantee the task quit. But block `Dispose` is not a ggod idea.